lua/neogit/lib/git/rebase.lua

```diff
@@ -58,6 +58,20 @@ function M.onto(start, newbase, args)
   end
 end
 
+function M.reword(commit, message)
```

Comment:
In the action you're passing the popup arguments as a third parameter to this function:
```lua
git.rebase.reword(commit, new_message or "", popup:get_arguments())
```

Presumably this can be used in the cli call on 67 to determine if autosquash/autostash should be enabled? Unless those are a hard-requirement for this to function properly.

Whichever way this needs to be changed is alright - but can you add type-annotations to this function?

```lua
---@param commit string
---@param message string
---@param arguments table
---@return nil
```

Reply:
Hmm, so `autosquash` is needed for sure for this to work, otherwise you get this "amend! ... " stuff as a separate commit.

Reply:
`autostash` is a different story. Technically this is not required. But during testing I found it a bit annoying to type `r w` and then get an error just because I had a dirty worktree. I think the reason why this is not a default in git is because the rebase can fail during the `git stash pop`. However since we do not change content during the rewording, this popping can really never fail AFAICS. I would personally leave it enabled by default like this, but if you say it should come from the popup args I'll refactor =)

Reply:
Nah, I think you made the right call. Leave it as is :)